zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/Learner.java

```diff
@@ -758,8 +759,9 @@ protected void syncWithLeader(long newLeaderZxid) throws Exception {
                         FollowerZooKeeperServer fzk = (FollowerZooKeeperServer) zk;
                         for (PacketInFlight p : packetsNotCommitted) {
                             fzk.logRequest(p.hdr, p.rec, p.digest);
+                            lastLoggedZxidFromDiff = p.hdr.getZxid();
                         }
-                        packetsNotCommitted.clear();
+                        fzk.getZKDatabase().commit();
```

Comment:
Hmm, need a rendevous with the sync processor thread for this to really guarantee anything.

Reply:
Fixed in next commit.